### PR TITLE
Logic migration

### PR DESCRIPTION
**Pull request by jooohhn**
### Work in Progress
- [x] Renamed App -> HomePage and created a new App component to wrap routes.
- [x] Necessary for nesting routes in react-router@4.*
- [x] Replace react-virtualized with handsontable
- [x] Re-enable progress bar
- [x] Table definition
- [x] Change falcon-core dep to `amilajack/falcon-core`
- [x] Query View
- [x] Structure View
- [x] Fix broken graph page
- [x] Migrate `graphql-voyager` to [lerna](https://github.com/lerna/lerna/)
- [x] Fix merge conflicts

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -14,5 +14,7 @@
   },
   "license": "MIT",
   "dependencies": {
+    "electron-store": "^1.2.0",
+    "falcon-core": "0.1.0-4"
```
Comment (amilajack) [inline app/package.json]: Change this to https://github.com/amilajack/falcon-ui-rewrite-2/blob/master/app/package.json#L18. Master is correct here

File: internals/scripts/CheckBuiltsExist.js [hunk 2]
```diff
@@ -9,15 +9,11 @@ function CheckBuildsExist() {
   const rendererPath = path.join(__dirname, '..', '..', 'app', 'dist', 'renderer.prod.js');
 
   if (!fs.existsSync(mainPath)) {
```
Comment (amilajack) [inline internals/scripts/CheckBuiltsExist.js]: Revert all the changes to the `internals` directory and any other boilerplate related files like webpack/babel/eslint changes. I have updated those in the repo already

**Comment by amilajack** [comment]
* screen shot 2018-01-31 at 2 55 25 pm
The font size of the buttons changed in the last commit

* Also the grid text should be left-aligned
* Also that scrolling perf is 🔥 🔥 🔥 . Nice work 🎉

**Comment by jooohhn** [comment]
Had some problems with overflow:scroll using handsontable. I moved the `<Footer>` component out of `<div className=".Grid">`.  Will fix before merge

**Comment by amilajack** [comment]
Also:

screen shot 2018-01-31 at 3 01 55 pm
rows should be white like so:
screen shot 2018-01-31 at 3 02 09 pm

**Comment by amilajack** [comment]
Some overflow issues with the dropdown in structure view (pretty sure you know about this). Also we have to find a way to convert the ints to booleans or their correct corresponding values. Also did you port the functionality for the "yellow null boxes"?

**Comment by amilajack** [comment]
Nit: Currently selected table should be highlighted in sidebar. Also gotta say. really impressed with how the changes are coming along 🎉

**Comment by amilajack** [comment]
screen shot 2018-02-03 at 3 57 50 pm
Something I forgot to add to the previous mockup: table definition. Should be an un-editable and syntax highlighted ace/brace component

**Comment by amilajack** [comment]
Nit: If the currently selected tab in the sidebar is a table then the "Graph" button should be hidden. It should only be shown when the currently selected tab is a database. "Structure" and "Content" buttons should also be hidden if current tab is a database

**Comment by amilajack** [comment]
screen shot 2018-02-05 at 1 52 25 pm

Type fields seem to have been broken

**Comment by amilajack** [comment]
screen shot 2018-02-05 at 3 54 17 pm
screen shot 2018-02-05 at 3 53 40 pm

Also there's some nits that need to be fixed in the login page. The connection should be listed in the sidebar when the user has not yet connected to a database yet. I just created an API in `falcon-core` that exposes this functionality. See these [falcon-core-example](https://github.com/amilajack/falcon-core-example/blob/master/sqlite-example.js#L19-L37) changes

**Comment by jooohhn** [comment]
My bad dude, I muted this. Will work on 
- [x] In Structure - `<Select>` overflow problem
- [x] In Structure - converting ints to booleans
- [ ] In Structure - Porting the 'NULL' feature
- [x] In Grid -Prohibit all multi-cell selection except in the case of multi-row selection

Edit this if there's anything else to be done before I start the CRUD commits

**Comment by amilajack** [comment]
screen shot 2018-02-10 at 8 18 08 pm

This table seems to have no type. Any idea why this is?

**Comment by jooohhn** [comment]
The content we're getting from falcon-core doesn't have a type. Probably because `sqlite_sequence` is a special internal table. Not sure what to do from here. Thoughts?

image

^^ the values we get logged from falcon-core

**Comment by amilajack** [comment]
screen shot 2018-02-10 at 10 00 43 pm

Checked this in db browser. Seems like an issue in core

**Comment by jooohhn** [comment]
I think either db-browser is using some type of substitution or maybe the return values of node-sqlite3 is different from whatever db-browser is using. These are the results from the original falcon and an online viewer:

image
image

**Comment by amilajack** [comment]
Nice! Good call

**Comment by amilajack** [comment]
What we're doing in `Database.js` is bad for performance. For each function call, we're creating a new database connection. Instead we should leverage a connection pool. Maybe one connection for each UI tab? Let's try to remove `Database.js` before merging this

**Comment by amilajack** [comment]
nit: Don't use underscores in filenames: 
screen shot 2018-03-03 at 12 14 06 pm

**Pull request merged**